orttraining/orttraining/core/session/training_session.h

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <memory>
+#include "core/common/optional.h"
+#include "core/common/path_string.h"
+#include "core/session/inference_session.h"
+#include "orttraining/core/graph/loss_func/loss_func_common.h"
+#include "orttraining/core/graph/loss_function_registry.h"
+#include "orttraining/core/graph/optimizer_graph_output_key.h"
+#include "orttraining/core/graph/optimizer_config.h"
+
+namespace onnxruntime {
+namespace training {
+
+class TrainingSession : public InferenceSession {
+ public:
+  typedef std::unordered_map<std::string /*OpType*/,
+                             std::vector<std::pair<size_t /*InputIndex*/, float /*value*/>>>
+      ImmutableWeights;
+
+  explicit TrainingSession(const SessionOptions& session_options,
+                           logging::LoggingManager* logging_manager = nullptr)
+      : InferenceSession(session_options, logging_manager) {}
```

Comment:
InferenceSession has changed its constructor in master, which means TrainingSession also needs to update before merging into master. I have a pr that fixes TrainingSession for the new constructor version. To minimize merge conflict and manual work, maybe you could consider pull that change once it is checked in.